### PR TITLE
Adding custom hostname commands to cli

### PR DESCRIPTION
**Pull request by mcherniak**
Adding custom hostname commands to cli.

TODO:

- Better error handling
- Success feedback for deletion
  - https://github.com/cloudflare/cloudflare-go/issues/126
- Pagnation (maybe)
- Add ability to set metadata

Examples;

`flarectl zone customhostname list --zone testzone.com`
`flarectl zone customhostname create --zone testzone.com --hostname www.testhost.com --ssl-method http --ssl-type dv`
`flarectl zone customhostname delete --zone testzone.com --hostname www.testhost.com`
`flarectl zone customhostname info --zone testzone.com --hostname www.testhost.com`

### CODE REVIEW

**Comment by elithrar** [comment]
@mcherniak - let me know when this is ready to review.

**Comment by mcherniak** [comment]
@elithrar Should be good for a first look

File: cmd/flarectl/flarectl.go [hunk 1]
```diff
@@ -156,6 +156,81 @@ func main() {
...
 				{
+					Name:    "customhostname",
+					Aliases: []string{"ch"},
+					Usage:   "Custom hostname information",
+					Subcommands: []cli.Command{
+						{
+							Name:    "list",
+							Aliases: []string{"l"},
+							Action:  zoneCustHostList,
+							Usage:   "List all the custom hostname on an account",
```
Comment (elithrar) [inline cmd/flarectl/flarectl.go]: s/hostname/hostnames and s/account/zone

File: cmd/flarectl/flarectl.go [hunk 2]
```diff
@@ -156,6 +156,81 @@ func main() {
...
+							Aliases: []string{"d"},
+							Action:  zoneCustHostDelete,
+							Usage:   "Delete a custom hostname",
+							Flags: []cli.Flag{
+								cli.StringFlag{
+									Name:  "zone",
+									Usage: "zone name",
+								},
+								cli.StringFlag{
+									Name:  "hostname",
```
Comment (elithrar) [inline cmd/flarectl/flarectl.go]: `hostname-id`

File: cmd/flarectl/flarectl.go [hunk 3]
```diff
@@ -156,6 +156,81 @@ func main() {
...
+									Usage: "host name",
```
Comment (elithrar) [inline cmd/flarectl/flarectl.go]: `0d89c70d-ad9f-4843-b99f-6cc0252067e9`

(example ID from https://api.cloudflare.com/#custom-hostname-for-a-zone-delete-a-custom-hostname-and-any-issued-ssl-certificates-)

File: cmd/flarectl/flarectl.go [hunk 4]
```diff
@@ -156,6 +156,81 @@ func main() {
...
+					Aliases: []string{"ch"},
+					Usage:   "Custom hostname information",
+					Subcommands: []cli.Command{
+						{
+							Name:    "list",
+							Aliases: []string{"l"},
+							Action:  zoneCustHostList,
+							Usage:   "List all the custom hostname on an account",
+							Flags: []cli.Flag{
+								cli.StringFlag{
```
Comment (elithrar) [inline cmd/flarectl/flarectl.go]: Need a pagination mechanism here

File: cmd/flarectl/zone.go [hunk 5]
```diff
@@ -196,3 +196,194 @@ func zoneRecords(c *cli.Context) {
...
+		cli.ShowSubcommandHelp(c)
+		return
+	}
+	zoneID, err := api.ZoneIDByName(zone)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	customhostnamesResult, _, err := api.CustomHostnames(zoneID, 0, cloudflare.CustomHostname{})
```
Comment (elithrar) [inline cmd/flarectl/zone.go]: Need to wire up pagination (I recognize it's on your to-do list) as most customers have hundreds-to-tens-of-thousands of these by default.

**Comment by elithrar** [comment]
Hey @mcherniak - were you still interested in finishing this off? I realize you probably have a ton of other things to do, but just following up for completeness.

**Comment by elithrar** [comment]
I've picked this up and will open a new PR with changes. I've also made some public API updates ahead of https://github.com/cloudflare/cloudflare-go/pull/138 being merged.

**Comment by elithrar** [comment]
See https://github.com/cloudflare/cloudflare-go/pull/159

**Pull request closed without merging**